### PR TITLE
Feature/dagster env vars in databricks jobs

### LINKED ISSUE #10305: dagster_databricks Automatically Set Dagster Cloud Environment Variables
### What's the use case?

When submitting a job to databricks, it would be nice to automatically set environment variables that are not sensitive but are required for the dagster project code to load, in particular, environment variables normally  set by Dagster Cloud automatically.

### Ideas of implementation

Based off of: https://github.com/dagster-io/dagster/pull/8685/files

It should be possible to check if Dagster Cloud env vars are set, and if they are, add them to this list  dict that gets set in databricks, the equivalent of a user manually adding:

```
 "env_variables": {
            "DAGSTER_CLOUD_IS_BRANCH_DEPLOYMENT": os.getenv("DAGSTER_CLOUD_IS_BRANCH_DEPLOYMENT", ""),
            "DAGSTER_CLOUD_DEPLOYMENT_NAME": os.getenv("DAGSTER_CLOUD_DEPLOYMENT_NAME", "")
        },
```

### Additional information

_No response_

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

### PR DESCRIPTION
**Pull request by zyd14**
### Summary & Motivation

closes #10305 

### How I Tested These Changes

Added unit tests for ensuring environment variables are scooped up correctly. Ran a manual test on a local deployment with a job live in Databricks to ensure environment variables were added.

### CODE REVIEW

**Comment by clairelin135** [comment]
Hi @zyd14! Thanks for filing this PR. This is all looking great to me--I've run the tests and there are some minor pylint errors:
```
[2022-11-14T20:12:29Z] ************* Module dagster_databricks_tests.test_databricks_pyspark_step_launcher
--
  | [2022-11-14T20:12:29Z] dagster_databricks_tests/test_databricks_pyspark_step_launcher.py:39:12: W0612: Unused variable 'i' (unused-variable)
  | [2022-11-14T20:12:29Z] dagster_databricks_tests/test_databricks_pyspark_step_launcher.py:55:12: W0612: Unused variable 'i' (unused-variable)
  | [2022-11-14T20:12:29Z] dagster_databricks_tests/test_databricks_pyspark_step_launcher.py:67:8: W0612: Unused variable 'system_vars' (unused-variable)
  | [2022-11-14T20:12:29Z] dagster_databricks_tests/test_databricks_pyspark_step_launcher.py:68:12: W0612: Unused variable 'i' (unused-variable)
```
would you mind making these changes? I can approve these changes afterwards

File: python_modules/libraries/dagster-databricks/dagster_databricks_tests/test_databricks_pyspark_step_launcher.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        assert env_vars.env_variables == correct_vars
+
+    def test_given_no_add_dagster_env_vars_no_system_vars_added(
+        self, mock_step_launcher_factory, monkeypatch
+    ):
+        vars_to_add = {"add": "this"}
+        test_launcher = mock_step_launcher_factory(
+            add_dagster_env_variables=False, env_variables=vars_to_add
+        )
+        for var in enumerate(DAGSTER_SYSTEM_ENV_VARS):
```
Comment (clairelin135) [inline python_modules/libraries/dagster-databricks/dagster_databricks_tests/test_databricks_pyspark_step_launcher.py:54]: Hm... I think this line is throwing an error. I think we probably need to take out the `enumerate` call?
```
[2022-11-14T22:42:05Z] =================================== FAILURES ===================================
--
  | [2022-11-14T22:42:05Z] _ TestCreateRemoteConfig.test_given_no_add_dagster_env_vars_no_system_vars_added _
  | [2022-11-14T22:42:05Z]
  | [2022-11-14T22:42:05Z] self = <dagster_databricks_tests.test_databricks_pyspark_step_launcher.TestCreateRemoteConfig object at 0x7fe234f219a0>
  | [2022-11-14T22:42:05Z] mock_step_launcher_factory = <function mock_step_launcher_factory.<locals>._mocked at 0x7fe0ae9f4ee0>
  | [2022-11-14T22:42:05Z] monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7fe0ae9aa520>
  | [2022-11-14T22:42:05Z]
  | [2022-11-14T22:42:05Z]     def test_given_no_add_dagster_env_vars_no_system_vars_added(
  | [2022-11-14T22:42:05Z]         self, mock_step_launcher_factory, monkeypatch
  | [2022-11-14T22:42:05Z]     ):
  | [2022-11-14T22:42:05Z]         vars_to_add = {"add": "this"}
  | [2022-11-14T22:42:05Z]         test_launcher = mock_step_launcher_factory(
  | [2022-11-14T22:42:05Z]             add_dagster_env_variables=False, env_variables=vars_to_add
  | [2022-11-14T22:42:05Z]         )
  | [2022-11-14T22:42:05Z]         for var in enumerate(DAGSTER_SYSTEM_ENV_VARS):
  | [2022-11-14T22:42:05Z] >           monkeypatch.setenv(var, f"{var}_value")
  | [2022-11-14T22:42:05Z]
  | [2022-11-14T22:42:05Z] dagster_databricks_tests/test_databricks_pyspark_step_launcher.py:55:
  | [2022-11-14T22:42:05Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
  | [2022-11-14T22:42:05Z] .tox/py39/lib/python3.9/site-packages/_pytest/monkeypatch.py:302: in setenv
  | [2022-11-14T22:42:05Z]     self.setitem(os.environ, name, value)
  | [2022-11-14T22:42:05Z] .tox/py39/lib/python3.9/site-packages/_pytest/monkeypatch.py:266: in setitem
  | [2022-11-14T22:42:05Z]     self._setitem.append((dic, name, dic.get(name, notset)))
  | [2022-11-14T22:42:05Z] /usr/local/lib/python3.9/_collections_abc.py:763: in get
  | [2022-11-14T22:42:05Z]     return self[key]
  | [2022-11-14T22:42:05Z] /usr/local/lib/python3.9/os.py:676: in __getitem__
  | [2022-11-14T22:42:05Z]     value = self._data[self.encodekey(key)]
  | [2022-11-14T22:42:05Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
  | [2022-11-14T22:42:05Z]
  | [2022-11-14T22:42:05Z] value = (0, 'DAGSTER_CLOUD_DEPLOYMENT_NAME')
  | [2022-11-14T22:42:05Z]
  | [2022-11-14T22:42:05Z]     def encode(value):
  | [2022-11-14T22:42:05Z]         if not isinstance(value, str):
  | [2022-11-14T22:42:05Z] >           raise TypeError("str expected, not %s" % type(value).__name__)
  | [2022-11-14T22:42:05Z] E           TypeError: str expected, not tuple
```

Comment (zyd14) [reply python_modules/libraries/dagster-databricks/dagster_databricks_tests/test_databricks_pyspark_step_launcher.py:54]: ugh sorry! I should've ran those tests again after I made those changes.

**Review by clairelin135 — APPROVED** [review]
LGTM!

**Pull request merged**